contracts/NFTFactory.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.7.0 <0.9.0;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract NFTFactory is ERC721Enumerable, Ownable, ERC721Burnable {
+    string public baseURI;
+    uint256 public cost = 0.01 ether;
+
+    mapping(address => uint8) private _accountsNFTCount;
+
+    event Log(address sender, string message);
+    event Received(address caller, uint amount, string message);
+    event Refunded(address sender, uint amount);
+
+    constructor(
+        string memory _name,
+        string memory _symbol
+    ) ERC721(_name, _symbol) {
+        emit Log(msg.sender, "constructor has called");
+        baseURI = "ipfs://QmaggY9Pq8CZNN4xRQKRAQKhAvtEzto2ndNXsqotsDyPM3/";
```

Comment:
optional - we can receive baseURI as constructor parameter instead of hardcoding.